### PR TITLE
[DRAFT] Support scalar BatchNormPatternFinder

### PR DESCRIPTION
**Pull request by llFreetimell**
This draft is for supporting BatchNormPatternFinder.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

---

Requested by @meejeong

### CODE REVIEW

**Comment by llFreetimell** [comment]
@jinevening Could you check whether I can merge this or not? :D (Not urgent!)

File: compiler/luci/pass/src/BatchNormPatternFinder.cpp [hunk 1]
```diff
@@ -44,10 +44,10 @@ bool is_batchnorm_add(const luci::CircleAdd *add, luci::CircleMul *&mul, luci::C
...
   }
 
-  if (constant->rank() != 1)
+  if (constant->rank() > 1)
     return false;
 
-  auto channel_dim = constant->dim(0);
+  auto channel_dim = constant->rank() == 1 ? constant->dim(0) : 1;
   // Assumption: Layout is channel-last
   if (!(channel_dim == add->dim(add->rank() - 1)))
```
Comment (jinevening) [inline compiler/luci/pass/src/BatchNormPatternFinder.cpp:52]: I think acceptable batchnorm const shapes are
1. rank 1 const whose channel size is the same with `add`
2. scalar const

This code may not handle the case when `add` has non-one channel.

Comment (llFreetimell) [reply compiler/luci/pass/src/BatchNormPatternFinder.cpp:52]: Oops, I missed that... thanks :D
"Handling scalar" is okay? :)

Comment (jinevening) [reply compiler/luci/pass/src/BatchNormPatternFinder.cpp:52]: I think it's okay. 

> Requested by @meejeong

I don't know what is the exact request, but the idea looks feasible to me.

Comment (llFreetimell) [reply compiler/luci/pass/src/BatchNormPatternFinder.cpp:52]: Ah, the request was "supporting scalar for `ReplaceMulAddWithDepthwiseConvPass`" :)
Current `ReplaceMulAddWithDepthwiseConvPass` only accepts rank-1 constant value but scalar can be accepted actually :D

**Comment by llFreetimell** [comment]
Re-open when this feature is really needed, close.

**Pull request closed without merging**